### PR TITLE
Initial work to get illuminance and occupancy into modernExtend

### PR DESCRIPTION
**Pull request by Crosenhain**
Attempting to modify modernExtend for use with [6974](https://github.com/Koenkk/zigbee-herdsman-converters/pull/6974)

Not yet tested, I am going to be setting up a test Z2M instance with a spare Zigbee radio to test this out.

### CODE REVIEW

**Comment by sjorge** [comment]
It's better to create a new illuminance extend that just calls the more generic numeric one and then have that one do the extra work to get the _lux variant too.

**Comment by Crosenhain** [comment]
I see, https://github.com/Koenkk/zigbee-herdsman-converters/blob/0fcaeef8210c6c2635abe85c37e2a26c6f17d4f3/src/converters/fromZigbee.ts#L448-L456 returns both illuminance and illuminance_lux so that should handle the conversion without any extra effort here. 

I still need to test this works as expected, I just haven't had a chance to set up the test environment.

File: src/lib/modernExtend.ts [hunk 1]
```diff
@@ -656,6 +656,32 @@ export function humidity(args?: Partial<NumericArgs>) {
     });
 }
 
+export function luminance(args?: Partial<NumericArgs>) {
```
Comment (mrskycriper) [inline src/lib/modernExtend.ts:659]: Provably better to call it `illuminance`

File: src/lib/modernExtend.ts [hunk 2]
```diff
@@ -656,6 +656,32 @@ export function humidity(args?: Partial<NumericArgs>) {
...
+        cluster: 'msIlluminanceMeasurement',
+        attribute: 'measuredValue',
+        reporting: {min: '10_SECONDS', max: '1_HOUR', change: 1},
+        description: 'Luminance level',
+        access: 'STATE_GET',
+        ...args,
+    });
+}
+
+export function occupancy(args?: Partial<BinaryArgs>) {
```
Comment (mrskycriper) [inline src/lib/modernExtend.ts:671]: What converter is this based on? It doesn't match current `occupancy`.

```js
    occupancy: {
        // This is for occupancy sensor that send motion start AND stop messages
        cluster: 'msOccupancySensing',
        type: ['attributeReport', 'readResponse'],
        options: [exposes.options.no_occupancy_since_false()],
        convert: (model, msg, publish, options, meta) => {
            if (msg.data.hasOwnProperty('occupancy')) {
                const payload = {occupancy: (msg.data.occupancy % 2) > 0};
                utils.noOccupancySince(msg.endpoint, options, publish, payload.occupancy ? 'stop' : 'start');
                return payload;
            }
        },
    } satisfies Fz.Converter,
```

**Comment by mrskycriper** [comment]
I propose adding new `ScaleFunction` as an alternative to numeric scaling.

Then in `illuminance` function two numerics will be created: `illuminance` and `illuminance_lux` with `luxScale` function. Their objects will be merged to create a combined `ModernExtend` and returned as a result.

I've already implemented them in code, but not sure, how to contribute it. I can either create a PR to this specific fork and try proposing a merge or I can take over the process and create a new PR.

@Crosenhain what do you think of this situation?

**Comment by Crosenhain** [comment]
Hey @mrskycriper, tbh I haven't had a chance to look at this further yet due to life events. 

Cheers for the comments in your review of my PR, I'll have to take a look at them. I am not too familiar with TS. 

Your solution sounds interesting, perhaps make another PR with that, and then I can rebase this PR if/when it is accepted?

**Comment by mrskycriper** [comment]
@Crosenhain got you, I'll make a separate PR then.

**Comment by Crosenhain** [comment]
@mrskycriper has done some amazing work in https://github.com/Koenkk/zigbee-herdsman-converters/pull/7083 that covers what this PR was going to do, and more!

**Pull request closed without merging**